hazelcast/src/main/java/com/hazelcast/memory/HeapMemoryManager.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.memory;
+
+import com.hazelcast.internal.memory.MemoryAccessor;
+import com.hazelcast.internal.memory.impl.EndiannessUtil;
+
+import java.util.Arrays;
+
+import static com.hazelcast.internal.memory.impl.EndiannessUtil.BYTE_ARRAY_ACCESS;
+
+/**
+ * Manages memory in Java byte arrays. For testing only.
+ */
+public class HeapMemoryManager implements MemoryManager {
+
+    private static final int HEAP_BOTTOM = 8;
+
+    private final Allocator malloc = new Allocator();
+
+    private final Accessor mem = new Accessor();
+
+    private final byte[] storage;
```

Comment:
This is not the only aspect in which this class is just a mock with minimum functionality. MemoryAccessor supports only `long` and `int` methods, `free` is a no-op, and so on. I can make `dispose()` behave, but it will not make this whole class anywhere close to usable outside of the narrow test context.